python/tests/conftest.py

```diff
@@ -112,3 +113,8 @@ def table_schema_nested():
 @pytest.fixture(scope="session", autouse=True)
 def foo_struct():
     return FooStruct()
+
+
+@pytest.fixture
+def catalog() -> InMemoryCatalog:
+    return InMemoryCatalog("test.in.memory.catalog", {"test.key": "test.value"})
```

Comment:
`test.key` isn't used anywhere, nor is `test.in.memory.catalog`. Can you add tests for properties and name?